frontends/p4/tableKeyNames.cpp

```diff
@@ -57,6 +72,9 @@ class KeyNameGenerator : public Inspector {
         name.emplace(expression, l + "[" + r + "]");
     }
 
+    /** Compute a name annotation for @expression.  Eg. `@name("bits")` for
+     * `bits & 0x3` or `0x3 & bits`.
```

Comment:
Note that this method is somewhat of a hack designed to support masks for keys, which are a P4-14 feature only.
I can argue that this should not be done here, but in the P4-14 front-end.